packages/babel-plugin-jest-hoist/src/index.js

```diff
@@ -107,17 +107,20 @@ FUNCTIONS.mock = args => {
       if (!found) {
         invariant(
           (scope.hasGlobal(name) && WHITELISTED_IDENTIFIERS[name]) ||
-          /^mock/.test(name) ||
-          // Allow istanbul's coverage variable to pass.
-          /^(?:__)?cov/.test(name),
+            /^mock/.test(name) ||
+            // Allow istanbul's coverage variable to pass.
+            /^(?:__)?cov/.test(name),
           'The module factory of `jest.mock()` is not allowed to ' +
-          'reference any out-of-scope variables.\n' +
-          'Invalid variable access: ' + name + '\n' +
-          'Whitelisted objects: ' +
-          Object.keys(WHITELISTED_IDENTIFIERS).join(', ') + '.\n' +
-          'Note: This is a precaution to guard against uninitialized mock ' +
-          'variables. If it is ensured that the mock is required lazily, ' +
-          'variable names prefixed with `mock` are permitted.',
+            'reference any out-of-scope variables.\n' +
```

Comment:
please open an issue on the prettier repo, it should be discussed, I can see pros and cons.